src/Mvc/test/Mvc.IntegrationTests/ActionParametersIntegrationTest.cs

```diff
@@ -458,8 +521,156 @@ public async Task ActionParameter_BindingToTypeWithNoParameterlessConstructor_Th
             Assert.Equal(
                 string.Format(
                     "Could not create an instance of type '{0}'. Model bound complex types must not be abstract or " +
-                    "value types and must have a parameterless constructor.",
-                    parameterType.FullName),
+                    "value types and must have a parameterless constructor. Record types must have a single primary constructor.",
+                    typeof(AbstractClassWithNoDefaultConstructor).FullName),
+                exception.Message);
+        }
+
+        public class ActionParameter_MultipleConstructors_ParameterlessConstructorModel
+        {
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel(string name) => (Name) = name;
+
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel(string name, int age) => (Name, Age) = (name, age);
+
+            public ActionParameter_MultipleConstructors_ParameterlessConstructorModel() { }
+
+            public string Name { get; set; }
+
+            public int Age { get; set; }
+        }
+
+        [Fact]
+        public async Task ActionParameter_MultipleConstructors_ParameterlessConstructor()
+        {
+            // Arrange
+            var parameterType = typeof(ActionParameter_MultipleConstructors_ParameterlessConstructorModel);
+            var parameterBinder = ModelBindingTestHelper.GetParameterBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "p",
+                ParameterType = parameterType
+            };
+            var testContext = ModelBindingTestHelper.GetTestContext(request =>
+            {
+                request.QueryString = QueryString.Create("Name", "James");
+            });
+            var modelState = testContext.ModelState;
+
+            // Act
+            var result = await parameterBinder.BindModelAsync(parameter, testContext);
+
+            // Assert
+            Assert.True(modelState.IsValid);
+
+            var model = Assert.IsType<ActionParameter_MultipleConstructors_ParameterlessConstructorModel>(result.Model);
+            Assert.Equal("James", model.Name);
+        }
+
+        public class ActionParameter_MultipleConstructors_NoParameterlessConstructorModel
+        {
+            public ActionParameter_MultipleConstructors_NoParameterlessConstructorModel(string name) => (Name) = (name);
+
+            public ActionParameter_MultipleConstructors_NoParameterlessConstructorModel(string name, int age) => (Name, Age) = (name, age);
+
+            public string Name { get; init; }
+
+            public int Age { get; init; }
+        }
+
+        [Fact]
+        public async Task ActionParameter_MultipleConstructors_NoParameterlessConstructor_Throws()
```

Comment:
Isn't this checked a few other ways❔ That is, this isn't checking a `record` type and it's unclear why the test was added.